tests/integration/test_issue_1656.py

```diff
@@ -0,0 +1,170 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import os
+import subprocess
+import sys
+
+import pytest
+
+from pex.interpreter import PythonInterpreter
+from pex.pex_info import PexInfo
+from pex.testing import make_env, run_pex_command
+from pex.typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Any, List, Text
+
+
+def create_pex_pex(
+    pex_version,  # type: str
+    pex_file,  # type: str
+):
+    # type: (...) -> None
+
+    run_pex_command(
+        args=["pex=={pex_version}".format(pex_version=pex_version), "-c", "pex", "-o", pex_file]
+    ).assert_success()
+
+
+@pytest.fixture(scope="module")
+def old_pex(tmpdir_factory):
+    # type: (Any) -> str
+
+    # This was the first version to support Python 3.10, and it did not rely upon the RECORD file to
+    # build venvs.
+    pex_version = "2.1.55"
+    pex_file = os.path.join(str(tmpdir_factory.mktemp(pex_version)), "tool.pex")
+    create_pex_pex(pex_version=pex_version, pex_file=pex_file)
+    return pex_file
+
+
+def run_current_pex_tool(
+    subject_pex,  # type: str
+    subcommand,  # type: str
+    *args  # type: str
+):
+    # type: (...) -> Text
+
+    return subprocess.check_output(
+        args=[sys.executable, "-m", "pex.tools", subject_pex, subcommand] + list(args)
+    ).decode("utf-8")
+
+
+def run_pex_tool(
+    pex_pex,  # type: str
+    subject_pex,  # type: str
+    subcommand,  # type: str
+    *args  # type: str
+):
+    # type: (...) -> Text
+    return subprocess.check_output(
+        args=[pex_pex, subject_pex, subcommand] + list(args), env=make_env(PEX_MODULE="pex.tools")
+    ).decode("utf-8")
+
+
+def test_old_venv_tool_vs_new_pex(
+    tmpdir,  # type: Any
+    old_pex,  # type: str
+):
+    # type: (...) -> None
+
+    pex_app = os.path.join(str(tmpdir), "app.pex")
+    run_pex_command(args=["cowsay==4.0", "-c" "cowsay", "-o", pex_app]).assert_success()
+
+    venv = os.path.join(str(tmpdir), "venv")
+    run_pex_tool(old_pex, pex_app, "venv", "--force", venv)
+    assert b"4.0\n" == subprocess.check_output(args=[os.path.join(venv, "pex"), "--version"])
+
+
+def test_new_venv_tool_vs_old_pex(
```

Comment:
This test requires `pex.venv.pex` falling back when it fails to `InstalledWheel.load(...)` fails. Fails without that.